apisix/secret/gcp.lua

```diff
@@ -0,0 +1,202 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+--- GCP Tools.
+local core       = require("apisix.core")
+local http       = require("resty.http")
+local google_oauth = require("apisix.utils.google-cloud-oauth")
+
+local sub        = core.string.sub
```

Comment:
```suggestion
local str_sub        = core.string.sub
```

only use `sub` is not a good name, we should keep `str_` or `string_` as prefix

Reply:
Fixed.